clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -257,8 +258,10 @@ static void validatePackoffset(Sema &S, HLSLBufferDecl *BufDecl) {
       VarDecl *NextVar = PackOffsetVec[i + 1].first;
       S.Diag(NextVar->getLocation(), diag::err_hlsl_packoffset_overlap)
           << NextVar << Var;
+      IsValid = false;
     }
   }
+  BufDecl->setHasPackoffset(IsValid);
```

Comment:
What is this for?  If I change this to `BufDecl->setHasPackoffset(true)` then the tests still pass.

Reply:
I have updated it to use `HasValidPackOffset`.